File: typecast-csharp/src/Typecast/Models/EmotionPreset.cs
using System.Text.Json.Serialization;

namespace Typecast.Models;

/// <summary>
/// Available emotion presets for TTS synthesis.
/// </summary>
[JsonConverter(typeof(JsonStringEnumMemberConverter))]
public enum EmotionPreset
{
    /// <summary>Neutral/normal tone</summary>
    [JsonPropertyName("normal")] Normal,
    
    /// <summary>Happy/cheerful tone</summary>
    [JsonPropertyName("happy")] Happy,
    
    /// <summary>Sad/melancholic tone</summary>
    [JsonPropertyName("sad")] Sad,
    
    /// <summary>Angry/frustrated tone</summary>
    [JsonPropertyName("angry")] Angry,
    
    /// <summary>Whispered/quiet tone</summary>
    [JsonPropertyName("whisper")] Whisper,
    
    /// <summary>Higher pitched/excited tone</summary>
    [JsonPropertyName("toneup")] ToneUp,
    
    /// <summary>Lower pitched/calm tone</summary>
    [JsonPropertyName("tonedown")] ToneDown
}

/// <summary>
/// Extension methods for EmotionPreset enum.
/// </summary>
public static class EmotionPresetExtensions
{
    /// <summary>
    /// Converts the EmotionPreset to its API string representation.
    /// </summary>
    public static string ToApiString(this EmotionPreset preset) => preset switch
    {
        EmotionPreset.Normal => "normal",
        EmotionPreset.Happy => "happy",
        EmotionPreset.Sad => "sad",
        EmotionPreset.Angry => "angry",
        EmotionPreset.Whisper => "whisper",
        EmotionPreset.ToneUp => "toneup",
        EmotionPreset.ToneDown => "tonedown",
        _ => throw new ArgumentOutOfRangeException(nameof(preset))
    };

    /// <summary>
    /// Parses a string to EmotionPreset enum.
    /// </summary>
    public static EmotionPreset ParseEmotionPreset(string value) => value.ToLowerInvariant() switch
    {
        "normal" => EmotionPreset.Normal,
        "happy" => EmotionPreset.Happy,
        "sad" => EmotionPreset.Sad,
        "angry" => EmotionPreset.Angry,
        "whisper" => EmotionPreset.Whisper,
        "toneup" => EmotionPreset.ToneUp,
        "tonedown" => EmotionPreset.ToneDown,
        _ => throw new ArgumentException($"Unknown emotion preset: {value}", nameof(value))
    };
}

File: typecast-java/src/main/java/io/typecast/models/EmotionPreset.java
package io.typecast.models;

/**
 * Emotion presets for TTS synthesis.
 * 
 * <p>SSFM v2.1 supports: normal, happy, sad, angry</p>
 * <p>SSFM v3.0 supports all presets including: whisper, toneup, tonedown</p>
 */
public enum EmotionPreset {
    /** Normal emotion (neutral) */
    NORMAL("normal"),
    /** Happy emotion */
    HAPPY("happy"),
    /** Sad emotion */
    SAD("sad"),
    /** Angry emotion */
    ANGRY("angry"),
    /** Whisper style (SSFM v3.0 only) */
    WHISPER("whisper"),
    /** Tone up style (SSFM v3.0 only) */
    TONEUP("toneup"),
    /** Tone down style (SSFM v3.0 only) */
    TONEDOWN("tonedown");

    private final String value;

    EmotionPreset(String value) {
        this.value = value;
    }

    /**
     * Returns the string value of the emotion preset.
     * 
     * @return the emotion preset string
     */
    public String getValue() {
        return value;
    }

    /**
     * Creates an EmotionPreset from a string value.
     * 
     * @param value the emotion preset string
     * @return the corresponding EmotionPreset
     * @throws IllegalArgumentException if the value doesn't match any preset
     */
    public static EmotionPreset fromValue(String value) {
        for (EmotionPreset preset : values()) {
            if (preset.value.equalsIgnoreCase(value)) {
                return preset;
            }
        }
        throw new IllegalArgumentException("Unknown emotion preset: " + value);
    }

    @Override
    public String toString() {
        return value;
    }
}
